packages/frontend/app/root.tsx

```diff
@@ -30,19 +36,37 @@ export const meta: MetaFunction = () => [
 ]
 
 export const loader = async ({ request }: LoaderFunctionArgs) => {
-  const session = await messageStorage.getSession(request.headers.get('cookie'))
+  const cookies = request.headers.get('cookie')
+  await redirectIfUnauthorizedAccess(request.url, cookies)
+
+  let logoutUrl = ''
+  const loggedIn = await isLoggedIn(cookies)
+  if (loggedIn) {
+    const response = await axios.get(
+      `${variables.kratosContainerPublicUrl}/self-service/logout/browser`,
+      {
+        headers: {
+          cookie: cookies
+        },
+        withCredentials: true
+      }
+    )
+    //TODO: In the scenario where a user is logged in but we hgave an issue retrieving the logoutUrl is it okay to simply not have a logoutUrl and not display the button? Given that the session will expire? Not sure how else to handle the error.
+    logoutUrl = response.data.logout_url
```

Comment:
> Given that the session will expire

When would it expire? Like on the next page visit which requires authorization action? I guess I don't fully understand the scenario, but could we make the logout button click just _look_ like they visited the logout link in this scenario (deleting a browser cookie or something)? Or should we force-logout if we know they are now unauthorized but are still seeing a logged in view until the next action?

I was actually wondering how the logout link works. I was thinking it would be nice if we didnt have to make this request in root (so on every page?) and either store it (pass it to the client to store somewhere?) or get the link on logout click from the client, depending on just how public the`kratosContainerPublicUrl` is (can we share it with the browser)?

Reply:
Looking at it further, the token/session are storedd in http only cookies so we cant delete them in the browser. We'd have to do it using Set-Cookie header and the cookie name (like we are doing just below here for the http only message cookie on line 72).

Reply:
The scenario I'm sketching here is just if we encounter an error in extracting the logout URL from the response data
```
const response = await axios.get(
      `${variables.kratosContainerPublicUrl}/self-service/logout/browser`,
      {
        headers: {
          cookie: cookies
        },
        withCredentials: true
      }
    )
    logoutUrl = response.data.logout_url
    ```
    
    Maybe we got an error back or there were network issues or whatever.